pkg/client/record/event_test.go

```diff
@@ -56,6 +61,57 @@ func (t *testEventSink) Update(e *api.Event) (*api.Event, error) {
 	return e, nil
 }
 
+// PatchEvent records the event for testing.
+func (t *testEventSink) Patch(e *api.Event, p []byte) (*api.Event, error) {
+	if t.OnPatch != nil {
+		return t.OnPatch(e, p)
+	}
+	return e, nil
+}
+
+type OnCreateFunc func(*api.Event) (*api.Event, error)
+
+func OnCreateFactory(testCache map[string]*api.Event, createEvent chan *api.Event) OnCreateFunc {
+	return func(event *api.Event) (*api.Event, error) {
+		// because the accuracy of time decreases to second-level after time is
+		// encoded, we have to manually shift the timestamp here, so the
+		// timestamp in patch will be different even if the same event occurs in
+		// rapid succession.
+		shift := time.Duration(-10 * time.Second)
+		event.FirstTimestamp = unversioned.NewTime(event.FirstTimestamp.Time.Add(shift))
```

Comment:
Where are these original timestamps generated? Somewhere in the code under test? And where does the 10 second shift matter? Is that comparison in the code under test as well? Or in the actual test code?

I don't know if it's possible, but I would rather see these just pull subsequent times off of a shared resource.